app/src/main/java/openfoodfacts/github/scrachx/openfood/views/LoadTaxonomiesService.java

```diff
@@ -56,45 +57,121 @@ private void doTask(Intent intent) {
-                .toCompletable()
-                //.doOnSubscribe(d -> view.showLoading())
-                 .subscribe(() -> {
-                    //view.hideLoading(false);
-                  //   builder.setContentText(getString(R.string.txtLoaded));
-                 //    builder.setOngoing(false);
-                 //    notificationManager.notify(17, builder.build());
-                }, e -> {
-                   //  e.printStackTrace();
-                    //view.hideLoading(true);
-                   //  builder.setContentText(getString(R.string.txtConnectionError))
-                    //         .setOngoing(false);
-                    // notificationManager.notify(17, builder.build());
-                 });
+        if (BuildConfig.FLAVOR.equals("off")) {
+            Single.zip(
+                    productRepository.getLabels(true),
+                    productRepository.getTags(true),
+                    productRepository.getAllergens(true),
+                    productRepository.getIngredients(false), //TODO : have a test on last-modified-date of ingredients.json before download it. Then pass the parameter to true
+                    productRepository.getCountries(true),
+                    productRepository.getAdditives(true),
+                    productRepository.getCategories(true), (labels, tags, allergens, ingredients, countries, additives, categories) -> {
+                        Completable.merge(
+                                Arrays.asList(
+                                        Completable.fromAction(() -> productRepository.saveLabels(labels)),
+                                        Completable.fromAction(() -> productRepository.saveTags(tags)),
+                                        Completable.fromAction(() -> productRepository.saveAllergens(allergens)),
+                                        Completable.fromAction(() -> productRepository.saveIngredients(ingredients)),
+                                        Completable.fromAction(() -> productRepository.saveCountries(countries)),
+                                        Completable.fromAction(() -> productRepository.saveAdditives(additives)),
+                                        Completable.fromAction(() -> productRepository.saveCategories(categories))
+                                )
+                        ).subscribeOn(Schedulers.computation())
+                                .subscribe(() -> {
+                                    settings.edit().putLong(Utils.LAST_REFRESH_DATE, System.currentTimeMillis()).apply();
+                                }, Throwable::printStackTrace);
+
+                        return true;
+                    })
+                    .subscribeOn(Schedulers.io())
+                    .observeOn(AndroidSchedulers.mainThread())
+                    .toCompletable()
+                    //.doOnSubscribe(d -> view.showLoading())
+                    .subscribe(() -> {
```

Comment:
please, remove commented line ( not in this PR but it will be great ;))

Reply:
For this one I've only duplicate the existing code and the commented one to be run function of flavors (OFF, OBF, ...).
Should I really removed this comments ?

Reply:
To be confirmed by @teolemon  or other reviewers but personnally I prefer to remove all commented code

Reply:
Commented code cleanup